File: benchmarks/tsb/bench_dataframe_expanding_sum_count.ts
/**
 * Benchmark: DataFrameExpanding.sum() and .count() on 10k-row DataFrame.
 * Outputs JSON: {"function": "dataframe_expanding_sum_count", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 10_000;
const WARMUP = 3;
const ITERATIONS = 10;

const a = Array.from({ length: ROWS }, (_, i) => (i % 100) * 1.5);
const b = Array.from({ length: ROWS }, (_, i) => (i % 50) * 2.0);
const df = DataFrame.fromColumns({ a, b });

for (let i = 0; i < WARMUP; i++) {
  df.expanding().sum();
  df.expanding().count();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  df.expanding().sum();
  df.expanding().count();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_expanding_sum_count",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_dataframe_expanding_sum_count.py
"""Benchmark: DataFrame.expanding().sum() and .count() on 10k-row DataFrame."""
import pandas as pd
import numpy as np
import json
import time

ROWS = 10_000
WARMUP = 3
ITERATIONS = 10

a = (np.arange(ROWS) % 100) * 1.5
b = (np.arange(ROWS) % 50) * 2.0
df = pd.DataFrame({"a": a, "b": b})

for _ in range(WARMUP):
    df.expanding().sum()
    df.expanding().count()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.expanding().sum()
    df.expanding().count()
total = time.perf_counter() - start

print(json.dumps({
    "function": "dataframe_expanding_sum_count",
    "mean_ms": total / ITERATIONS * 1000,
    "iterations": ITERATIONS,
    "total_ms": total * 1000,
}))
